File: 1605-minimum-number-of-days-to-make-m-bouquets/minimum-number-of-days-to-make-m-bouquets.cpp
class Solution {
public:
    int minDays(vector<int>& a, int m, int k) {
        int low=*min_element(a.begin(),a.end());
        int high=*max_element(a.begin(),a.end());
        int ans=-1;
        while(low<=high)
        {
            int mid=low+(high-low)/2;
            int consec=0;
            int bouq=0;
            for(int i=0;i<a.size();i++)
            {
                if(a[i]<=mid)
                {
                    consec++;
                
                if(consec==k)
                {
                    consec=0;
                    bouq++;
                }}
                else
                {
                    consec=0;
                }

            }

            if(bouq>=m)
            {
                ans=mid;
                high=mid-1;
            }
            else
            low=mid+1;
            
        }
        return ans;
    }
};

File: 1605-minimum-number-of-days-to-make-m-bouquets/minimum-number-of-days-to-make-m-bouquets.py
class Solution:
    def minDays(self, a, m, k):
        if m * k > len(a):
            return -1

        low = min(a)
        high = max(a)
        ans = -1

        while low <= high:
            mid = (low + high) // 2
            consec = 0
            bouq = 0

            for x in a:
                if x <= mid:
                    consec += 1
                    if consec == k:
                        bouq += 1
                        consec = 0
                else:
                    consec = 0

            if bouq >= m:
                ans = mid
                high = mid - 1
            else:
                low = mid + 1

        return ans


#__import__("atexit").register(lambda: open("display_runtime.txt", "w").write("0"))
